core/src/main/java/org/elasticsearch/common/Randomness.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common;
+
+import org.elasticsearch.bootstrap.BootstrapInfo;
+import org.elasticsearch.common.settings.Settings;
+
+import java.security.SecureRandom;
+import java.util.Random;
+
+public final class Randomness {
+    private static final SecureRandom SR = new SecureRandom();
+
+    private Randomness() {}
+
+    public static Random create(Settings settings, String setting) {
+        Long maybeSeed = settings.getAsLong(setting, null);
+        if (maybeSeed != null) {
+            return createFromSeed(maybeSeed);
+        } else {
+            return create();
+        }
+    }
+
+    public static Random create() {
+        String property = null;
+        try {
+            property = (String)BootstrapInfo.getSystemProperties().get("tests.seed");
+        } catch (SecurityException ignored) {
+            // getting system properties requires wildcard read-write
+        }
+        if (property != null) {
+            long accumulator = 0;
```

Comment:
Because [`Long#decode`](https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#decode-java.lang.String-) will throw an exception on some of the test seeds created by randomizedtesting. The key is this clause:

> This sequence of characters must represent a positive value or a NumberFormatException will be thrown.

This means that it throw an exception on any sequence of hexadecimal digits outside of `[0x0000000000000000--0x7FFFFFFFFFFFFF]`, .e.g., [0xCAFEBABE](https://en.wikipedia.org/wiki/Java_class_file#Magic_Number).